pubpub/pubpub
# dev(ts-migrate): remove babel (3/3)

This PR fully removes babel from our test and build systems.

- The `dist` directory now contains two child build directories, `client` and `server`.
- All server code was migrated to TypeScript.
- Babel extensions for jest (babel-jest) and Webpack (babel-loader) were swapped out for their TS counterparts (ts-jest and ts-loader, respectively).
- The NPM scripts were updated to automatically rebuild the server when changed. This is sped up significantly with incremental compilation.
- The workers were also migrated to TypeScript due to their coupled nature with the API server. The main worker processis executed with `ts-node` and thew workers it spawns are automatically run with `ts-node/register`. Watching/reloading in dev mode was tested as well as the following task-related items:
  * Verified HTML export SCSS->CSS compilation still works.
  * Tested a handful of export templates like LaTeX and JATS XML.

_Test Plan_
We will likely want to do a fairly comprehensive smoke test of the application before merging this into `master`, but for now, I'd suggest:
- Make sure to `npm install`.
- Verify the app launches and automatic reloading still works in dev mode.
- Verify the prod builds work.
- Verify imports and exports work.
- Ensure all tests pass.
- Launch Storybook and make sure nothing looks too suspicious.
